src/k8s/pkg/k8sd/api/cluster.go

```diff
@@ -22,7 +23,13 @@ var k8sdCluster = rest.Endpoint{
 }
 
 func clusterGet(s *state.State, r *http.Request) response.Response {
-	status, err := impl.GetClusterStatus(r.Context(), s)
+	var req apiv1.GetClusterStatusRequest
+	err := json.NewDecoder(r.Body).Decode(&req)
+	if err != nil {
+		return response.SmartError(fmt.Errorf("failed to decode request data: %w", err))
+	}
+
+	status, err := impl.GetClusterStatus(r.Context(), s, req.WaitReady)
```

Comment:
An issue with this is that the client that does `k8s status --wait-ready` will block while waiting on the server, and the server will then block waiting on something else.

I am not very fond of this approach, as it can easily backfire. A different approach could be that the server returns a status field `Ready: true/false` then the client retries the request if not. This has much clearer semantics and does not keep long-running server requests.

Reply:
I went with this approach as otherwise we would need to reimplement the `wait-ready` logic in the charm and other potential clients that use the REST API.
We don't want to have any business logic in the CLI - relying on the client to handle the `wait-ready` would be against that.

If you really feel we should go with the `Ready: true/false` approach, then this is fine for me and I'll change it. Just wanted to explain why I went down this path.

Reply:
I see the reasoning, indeed. However, the logic should just be a retry loop, so i still believe this is worth doing:

```
while !status.is_ready():
    status = client.GetStatus()
```

We should avoid long running RPCs unless we absolutely have to, since handling timeouts can be tricky (the client does not know where the timeout is, is it k8sd or the upstream service?)

Reply:
Alright, makes sense - I refactored it.